pkgs/os-specific/linux/firmware/ath9k/default.nix

```diff
@@ -0,0 +1,147 @@
+{ lib
+, stdenv
+, fetchurl
+, fetchFromGitHub
+, m4
+, cmake
+, perl
+, writeScript
+, enableUnstable ? false
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ath9k-htc-blobless-firmware";
+  version = if enableUnstable then "unstable-2022-05-22" else "1.4.0";
+
+  src = fetchFromGitHub ({
+    owner = "qca";
+    repo = "open-ath9k-htc-firmware";
+  } // (if enableUnstable then {
+    rev = "d856466a068afe4069335257c0d28295ff777d92";
+    hash = "sha256-9OE6qYGABeXjf1r/Depd+811EJ2e8I0Ni5ePHSOh9G4=";
+  } else {
+    rev = finalAttrs.version;
+    hash = "sha256-Q/A0ryIC5E1pt2Sh7o79gxHbe4OgdlrwflOWtxWSS5o=";
+  }));
```

Comment:
Inlining multiple versions is not that readable. A generic approach or an override is preferred.

Reply:
Okay, see 5b0d8e8b16c2a39f02b2d2edc96b9552fb1234d7; is that what you meant?

If so I will push an additional commit that moves the (large) `let common = ...;` into a separate `common.nix`.  But that will make the diffs harder to read so let's clear up all the other issues and then I'll do that right before the final squash+merge.